src/lib/operators/join_sort_merge.cpp

```diff
@@ -226,6 +259,179 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
   // Represents the result of a value comparison.
   enum class CompareResult { Less, Greater, Equal };
 
+  // In the case of the anti-join, we need to check that there is no combination where all predicates are satisfied. In
+  // the case of the semi-join, we need to check that there is at least one combination where all predicates are
+  // satisfied.
+  void _emit_multipredicate_semi_anti_null_as_false(size_t output_cluster, TableRange left_range,
+                                                    TableRange right_range,
+                                                    MultiPredicateJoinEvaluator& multi_predicate_join_evaluator) {
+    left_range.for_every_row_id(_sorted_left_table, [&](RowID left_row_id) {
+      auto match = false;
+      right_range.for_every_row_id(_sorted_right_table, [&](RowID right_row_id) {
+        if (multi_predicate_join_evaluator.satisfies_all_predicates(left_row_id, right_row_id)) {
+          match = true;
+          return;
+        }
+      });
+      if ((match && _mode == JoinMode::Semi) || (!match && _mode == JoinMode::AntiNullAsFalse)) {
+        _emit_left_only(output_cluster, left_row_id);
+      }
+    });
+  }
+
+  void _emit_multipredicate_anti_null_as_true(size_t output_cluster, TableRange left_range, TableRange right_range,
```

Comment:
How about the following comment:

```
  // In the case of the anti null as true join, we need to make sure that there is no match between the rows of the
  // left range and the NULL rows from the right table where all predicates are satisfied. In some cases, we also need
  // to check that there is no match with the rows from the right range. That is for example the case if the right and
  // left range have the same value because than there still can be a match for the current row where all predicates
  // are satisfied. If that is the case check_right_range should be set.
```